diff --git a/src/krabby/extension/extension.js b/src/krabby/extension/extension.js
index c39cea1..e2ec9be 100644
--- a/src/krabby/extension/extension.js
+++ b/src/krabby/extension/extension.js
@@ -172,8 +172,6 @@ function KrabbyExtension(krabby) {
   // Switch tabs
   krabby.modes.modal.map('Page', ['Alt', 'KeyL'], () => krabby.extensions.commands.send('next-tab'), 'Next tab', 'Switch tabs')
   krabby.modes.modal.map('Page', ['Alt', 'KeyH'], () => krabby.extensions.commands.send('previous-tab'), 'Previous tab', 'Switch tabs')
-  krabby.modes.modal.map('Page', ['Alt', 'Digit1'], () => krabby.extensions.commands.send('first-tab'), 'First tab', 'Switch tabs')
-  krabby.modes.modal.map('Page', ['Alt', 'Digit0'], () => krabby.extensions.commands.send('last-tab'), 'Last tab', 'Switch tabs')
 
   // Move tabs
   krabby.modes.modal.map('Page', ['Alt', 'Shift', 'KeyL'], () => krabby.extensions.commands.send('move-tab-right'), 'Move tab right', 'Move tabs')
